Source/SoundStarter_Patch.cs

```diff
@@ -0,0 +1,63 @@
+﻿using RimWorld.Planet;
+using System;
+using Verse;
+using Verse.Sound;
+
+namespace RimThreaded
+{
+    class SoundStarter_Patch
+	{
+		public static bool PlayOneShot(SoundDef soundDef, SoundInfo info)
+		{
+			if (UnityData.IsInMainThread)
+			{
+				return true;
+			}
+
+			if (soundDef == null)
+			{
+				Log.Error("Tried to PlayOneShot with null SoundDef. Info=" + info, false);
+				return false;
+			}
+			DebugSoundEventsLog.Notify_SoundEvent(soundDef, info);
+			if (soundDef == null)
+			{
+				return false;
+			}
+			if (soundDef.isUndefined)
+			{
+				return false;
+			}
+			if (soundDef.sustain)
+			{
+				Log.Error("Tried to play sustainer SoundDef " + soundDef + " as a one-shot sound.", false);
+				return false;
+			}
+			for (int i = 0; i < soundDef.subSounds.Count; i++)
+			{
+				Ticklist_Patch.PlayOneShot.Enqueue(new Tuple<SoundDef, SoundInfo>(soundDef, info));
+			}
+			return true;
```

Comment:
I have no idea why but when this is true it all works. Setting this to false causes threads to timeout.